File: idragonfly/Python Codes/dfconvect.py
import jax.numpy as jnp

def dfconvect(ZF, VELF, dt, iGAMAf):
    
    ZW = jnp.zeros((2, iGAMAf[0]))
    
    for iwing in range(0, 1):
        for i in range(iGAMAf[0]):
            ZW = ZW.at[iwing, i].set(ZF[iwing, i] + VELF[iwing, i] * dt)

    return ZW

File: idragonfly/MATLAB Codes/dfconvect.m
function [ZW]=dfconvect(ZF,VELF,dt,iGAMAf)
%Convect vortex from ZF to ZW using the velocity VELF*df
%INPUT
% ZF    location of the vortex
% VELF  velocity at the vortex
% dt    time interval
% iGAMAf    # of vortices to be convected
%OUTPUT
% ZW    location of the vortex after convection
    for iwing=1:2
    for i=1:iGAMAf(1)
        ZW(iwing,i)=ZF(iwing,i)+VELF(iwing,i)*dt;
    end
    end


end

